src/zitadel.rs

```diff
@@ -41,434 +48,256 @@ impl Zitadel {
-									retry_error
-								);
-							}
-						}
-					}
-				}
-			}
+		if self.feature_flags.is_enabled(FeatureFlag::DryRun) {
+			tracing::warn!("Skipping deletion due to dry run");
+			return Ok(());
 		}
 
-		Ok(())
+		self.zitadel_client.delete_user(zitadel_id).await.map(|_o| ())
 	}
 
-	/// Update a Zitadel user
-	async fn update_user(&self, old: &ZitadelUser, new: &ZitadelUser) -> Result<()> {
+	/// Import a user into Zitadel
+	pub async fn import_user(&mut self, imported_user: &User) -> Result<()> {
+		tracing::info!("Importing user with external ID: {}", imported_user.external_user_id);
+
 		if self.feature_flags.is_enabled(FeatureFlag::DryRun) {
-			tracing::info!("Not updating user due to dry run: {:?} -> {:?}", old, new);
+			tracing::warn!("Skipping import due to dry run");
 			return Ok(());
 		}
 
-		let Some(user_id) = self.get_user_id(old).await? else {
-			bail!("could not find user `{}` to update", old.user_data.email);
-		};
+		let mut metadata =
+			vec![SetMetadataEntry::new("localpart".to_owned(), imported_user.famedly_uuid())];
 
-		if old.user_data.email != new.user_data.email {
-			self.zitadel_client
-				.update_human_user_name(
-					&self.zitadel_config.organization_id,
-					user_id.clone(),
-					new.user_data.email.clone().to_string(),
-				)
-				.await?;
+		if let Some(preferred_username) = imported_user.preferred_username.clone() {
+			metadata
+				.push(SetMetadataEntry::new("preferred_username".to_owned(), preferred_username));
+		}
 
-			tracing::warn!(
-				"User email/login changed for {} -> {}",
-				old.user_data.email,
-				new.user_data.email
+		let mut user = AddHumanUserRequest::new(
+			SetHumanProfile::new(imported_user.first_name.clone(), imported_user.last_name.clone())
+				.with_nick_name(imported_user.external_user_id.clone())
+				.with_display_name(imported_user.get_display_name()),
+			SetHumanEmail::new(imported_user.email.clone())
+				.with_is_verified(!self.feature_flags.is_enabled(FeatureFlag::VerifyEmail)),
+		)
+		.with_organization(
+			Organization::new().with_org_id(self.zitadel_config.organization_id.clone()),
+		)
+		.with_metadata(metadata);
+
+		if let Some(phone) = imported_user.phone.clone() {
+			user.set_phone(
+				SetHumanPhone::new()
+					.with_phone(phone.clone())
+					.with_is_verified(!self.feature_flags.is_enabled(FeatureFlag::VerifyPhone)),
 			);
 		};
 
-		if old.user_data.first_name != new.user_data.first_name
-			|| old.user_data.last_name != new.user_data.last_name
-		{
-			self.zitadel_client
-				.update_human_user_profile(
-					&self.zitadel_config.organization_id,
-					user_id.clone(),
-					new.user_data.first_name.clone().to_string(),
-					new.user_data.last_name.clone().to_string(),
-					None,
-					Some(new.get_display_name()),
-					None,
-					None,
-				)
-				.await?;
-		};
+		if self.feature_flags.is_enabled(FeatureFlag::SsoLogin) {
+			user.set_idp_links(vec![IdpLink::new()
+				.with_user_id(imported_user.external_user_id.clone())
```

Comment:
We need to use the broken base64-encoded scheme here for Zitadel to interpret the IDs "correctly".

We should probably also send an issue upstream about this?

Reply:
Need a test to confirm this is correctly encoding IDs too.

Reply:
This is complex to write since openLDAP does not support binary IDs.

We should probably reconsider how the test suite handles IDs together with #97